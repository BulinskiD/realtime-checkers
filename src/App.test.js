import React from 'react';
import ConnectedApp, {App} from './App';
import Login from './auth';
import configureMockedStore from 'redux-mock-store';
import Authenticated from './shared/authenticated';
import ListGames from './list-games';
import Board from './board';
import {shallow, mount} from 'enzyme';


const onUserAuthChange = jest.fn();
let props;
const mockedStore = configureMockedStore();

describe("App", () => {
    beforeEach(() => {
        props = {authenticated: true, loading: false, onUserAuthChange};
        onUserAuthChange.mockClear();
    });

   it('Should render correct routes when loading is false', () => {
       const component = shallow(<App {...props} />);
       const listGamesFunc = component.find('Route').at(0).props().component();
       expect(listGamesFunc).toStrictEqual(<Authenticated authenticated={true} display={<ListGames />} />);

       const boardFunc = component.find('Route').at(1).props().component();
       expect(boardFunc).toStrictEqual(<Authenticated authenticated={true} display={<Board />} />);


       const comp = component.find('Route').at(2).props().component();
       expect(comp).toStrictEqual(<Login />);

       expect(component).toMatchSnapshot();
   });

    it('Should render loading when loading prop is true', () => {
        props.loading = true;
        const component = shallow(<App {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('Should call onUserAuthChange on load', () => {
        mount(<App {...props} />);
        expect(onUserAuthChange).toBeCalledTimes(1);
    });
});

describe('MapStateToProps', () => {
   it('Should map state to props correctly', () => {
      const store = mockedStore({user: {isLoggedIn: false, initial: false}});
      const component = shallow(<ConnectedApp store={store} />);
      expect(component.find('App').props().authenticated).toBe(false);
      expect(component.find('App').props().loading).toBe(false);
   });
});