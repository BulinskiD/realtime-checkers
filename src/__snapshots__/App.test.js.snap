// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render correct routes when loading is false 1`] = `
<BrowserRouter>
  <Container
    fluid={false}
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/game/:id"
    />
    <Route
      component={[Function]}
      path="/login"
    />
  </Container>
</BrowserRouter>
`;

exports[`App Should render loading when loading prop is true 1`] = `
<div>
  Ladowanie
</div>
`;
